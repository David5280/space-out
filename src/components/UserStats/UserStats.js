import React from 'react'
import { Link } from 'react-router-dom'

const UserStats = ({ planet, user }) => {

  const userStats = (
    <section className='user-stats'>
    <div className='user-stats-background-container'>
      <img 
        src={planet.image2} 
        alt={`${planet.title} background`} 
        className={`user-stats-background ${planet.title}-background`}
      />
    </div>
    <article className='user-stats-container'>
      <h3>{planet.title}</h3>
      <p>
        You would weigh {(user.weight * parseFloat(planet.gravity)).toFixed(2)} pounds on the surface of {planet.title}
      </p>
      <p>
        If you left today, you would be {parseInt(user.age) + parseFloat(planet.travelTime)} years old upon arrival
      </p>
    </article>
  </section>
  );

  const noUserStats = (
    <section className='no-credentials-page'>
      <Link to='/enter-stats'>Please enter your information to view this page.</Link>
    </section>
  );

  const userStatsDisplay = user.name && user.age && user.weight ? 
    userStats : 
    noUserStats;

  return (
    userStatsDisplay
  )
};

export default UserStats;