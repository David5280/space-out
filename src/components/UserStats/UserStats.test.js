import React from 'react';
import { shallow } from 'enzyme';
import UserStats from './UserStats';

describe('UserStats', () => {
  let mockPlanet;
  let mockUser;
  let mockUserEmpty;
  beforeEach(() => {
  })
  mockPlanet = {
    title: 'mars',
    image2: '/image2',
    gravity: 0.5,
    travelTime: 2
    }
  mockUser = {
      name: 'David',
      age: 25,
      weight: '180'
    }
  mockUserEmpty = {
      name: 'David',
      age: 25,
      weight: '180'
    }
  it('should match the snapshot with a user', () => {
    const wrapper = shallow(<UserStats planet={mockPlanet} user={mockUser} />)
    expect(wrapper).toMatchSnapshot();
  });
});