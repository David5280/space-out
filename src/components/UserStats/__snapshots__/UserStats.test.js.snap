// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserStats should match the snapshot with a user 1`] = `ShallowWrapper {}`;
